Newsweek Homepage Clone - Bootstrap

sneeu-leeu opened the pull request:
## Newsweek Website Clone

#### In This Project we:
* Setup project dependencies like GitHub Actions and Linters via Microverse Linters Config. 
* Defined a well structured DOM with each element properly grouped using proper sizes and alignments. 
* Used the browser's console to review the page's development strategy. 
* Used semantic tags and others were necessary.
* Primarily styled using Bootstrap

sneeu-leeu committed:
Add page dependancies

sneeu-leeu committed:
Add logo to repo

Ceejayski committed:
Add header tag to the body and styled it

Ceejayski committed:
Format CSS file

Ceejayski committed:
Add medium navbar and css

sneeu-leeu committed:
Add bottom large nav

sneeu-leeu committed:
Add xl-grid setup and style

sneeu-leeu committed:
Add debate and opinion section

Ceejayski committed:
Add first section and style it

sneeu-leeu committed:
Add bottom grid container

Ceejayski committed:
Fix container width

Ceejayski committed:
Add new article sections to the body

sneeu-leeu committed:
Add all sizes editors pick

Ceejayski committed:
Add footer to the body and it style

sneeu-leeu committed:
Add bottom grid

sneeu-leeu committed:
Complete content

Ceejayski committed:
Finish project and fix linter errors

Ceejayski committed:
Update README.md

Ceejayski committed:
Create screenshot.png

Ceejayski committed:
Update README.md

sneeu-leeu committed:
Fix offset error

sneeu-leeu committed:
Merge branch 'feature-a' of github.com:sneeu-leeu/Newsweek-bootstrap into feature-a

sneeu-leeu committed:
Update Readme

sneeu-leeu committed:
Update README.md

javitocor reviewed (changes requested):
#[CHANGES REQUIRED ♻️ ]

Good job so far, team! 👍 
Your code is clean and you used less than 300 lines in CSS, AWESOME! 🙈 
Good README file and descriptive pull request and commits! 💯 

However, the project needs some changes to be approved, check my comments below. 👇 

Make the necessary changes and request another code review when done

Good Luck 😃 

Reach me on Slack if you have any doubt or question, I will answer you as soon as possible 💬 

## Changes Required:

- In the original website, the first breakdown is at 990px, yours is at 1199px. Fix that!
![Original](https://user-images.githubusercontent.com/56809963/93371096-187be180-f852-11ea-94b9-6b79fd564300.PNG)
![Project](https://user-images.githubusercontent.com/56809963/93371101-1a45a500-f852-11ea-8af8-bc0a4fa33ce8.PNG)

- The subscribe and newsletter section needs to be fixed for different sizes:
![original](https://user-images.githubusercontent.com/56809963/93371593-e4ed8700-f852-11ea-931e-3cdfca240a02.PNG)
![project](https://user-images.githubusercontent.com/56809963/93371597-e6b74a80-f852-11ea-95a1-d8dc49dfd75e.PNG)
![project](https://user-images.githubusercontent.com/56809963/93371600-e74fe100-f852-11ea-899e-de3a72d14844.PNG)

- The navbar should be sticky

sneeu-leeu commented:
Thank you so much for your review, Javier! A detailed eye is always appreciated. However, the breakpoints used are standard from bootstrap as the project dictates and not from Media queries.

#### To our next review:

### We have:
* Fixed the small navbar to the top
* fixed the newsletter section to not break at medium (tablet)
* Moved the Subscribe section to overlap on the small screen with Z-index while trying not to use any further CSS

sneeu-leeu committed:
Add fixed-top nav

jubaan reviewed (changes requested):
# Hi team!! Awesome job :ok_hand:! 

### **I'm impress** :heart_eyes: with the great work you have done with the styling. I must say I sometimes got confuse :confused: about which one was the original and which one was the clone. 

:clap::clap::clap: Great job with this things :clap::clap::clap:: 
- :100:  Using **reset CSS**, and positioning at the beginning of the CSS stylesheet
- :100:  Seting up **Bootstrap** with CDN links for both the framework and the JS.
- :100:  Choosing **CSS classes** over CSS ID's
- :100: Using **GitHub Flow***
- :100:  Use of semantic HTML with tags like ```nav```, ```header```, ```main```, and ```footer``` just to mention some.
- :100:  Informative README, and descriptive PR, and Commit messages.
- :100:  Linters set-up correctly
- :100:  Passes all linters check

While it's a good :sunglasses: looking project still the code hasn't fix all previous observations, please work on them to move forward :see_no_evil:.

## Required changes

### [Mandatory]

- [x] Clone resposivenes doesn't match Original. **READ NOTES** at the end.
#### ORIGINAL
![image](https://user-images.githubusercontent.com/25774518/93412788-7cf36c80-f863-11ea-8772-bd9d7be43921.png)
![image](https://user-images.githubusercontent.com/25774518/93415108-58e65a00-f868-11ea-9b99-f233ab45ddd3.png)

#### CLONE
![image](https://user-images.githubusercontent.com/25774518/93412766-6f3de700-f863-11ea-92ff-fa4764c0fb6e.png)
![image](https://user-images.githubusercontent.com/25774518/93415071-45d38a00-f868-11ea-80c6-8a9ff667653a.png)

- [x] Remove the unnecessary files from the repository or add them to ```.gitignore``` file.
    ``` extra.html```
![image](https://user-images.githubusercontent.com/25774518/93410111-fe480080-f85d-11ea-85ea-0296aaa12b18.png)

### [OPTIONAL]

- [x] **Use descriptive names for feature branch**. Example:
    :-1:  ```feature-a``` is ok. But makes the reader question: What does it contain? What is it about? Where is ```feature-b```?
    :+1:  ```feature_newsweek-clone``` is better. Is still short and descriptive enough for the reader to grasp the main purpose or idea of the branch.
    Other examples but not limited: ``` newsweek-clone```, ``` newsweek-welcome-page-clone```, ```feature-main-newsweek-page```

- [x] **Have a separate reset css style-sheet**, and separate it from the main style-sheet.
    It helps maintain the CSS tidy and unclutter. It makes it easier to maintain in the long run.

- [x] ```border: none``` property could have its own CSS class or better yet the code could take advantage of Bootstrap ```border-0```.  [Bootstrap borders](https://getbootstrap.com/docs/4.0/utilities/borders/). This helps maintain the code DRY.

- [x] Prefer CSS selectors that are specific enough and easy to identify over general, non-specific selectors.

    :-1:  While this works, the reader could have difficulties identifying the html element is referring to. 
    ```
    nav div a {
      margin-top: 3px;
    }
    ```

    :+1:  This would be a better option for the code. Descriptive and easy to identify in the HTML file.
    ```
    nav__link {
      margin-top: 3px
     }
    ...
- [x] The **navbar** still is not sticky in all the views, only on mobile version.
#### ORIGINAL
![Newsweek website](https://user-images.githubusercontent.com/25774518/93411715-664c1600-f861-11ea-837c-22932b8cafc8.png)
#### CLONE
![clone](https://user-images.githubusercontent.com/25774518/93411762-7cf26d00-f861-11ea-89e0-0be7454d2e8f.png)

### [NOTES]

#### Responsiveness: 

This is tricky in this project and is hard to get right in the first try, but don't let this demotivate you, **because you have done amazing!**, is not easy for a lot of students get where you have. This is just a little bump in the road, so to help you figure it out and compare the clone against the original, use the **MEDIA QUERIES INSPECTOR** that comes with every browser. 

1- Open **Inspector Mode** in your web browser, most of them use ```Control``` + ```Shift``` + ```I```

2- Select the ```responsiveness``` option, it should look something like this...
![image](https://user-images.githubusercontent.com/25774518/93413632-4d456400-f865-11ea-9aa2-d84455393e47.png)

3- Start playing with the media queries by clicking on the gray areas
![image](https://user-images.githubusercontent.com/25774518/93413800-a9a88380-f865-11ea-8292-2b697e84a879.png)

4- Use the Bootstrap ones that are most similar or as last resort, as I have seen the code is using, modify the bootstrap original queries.. [Bootstrap layouts](https://getbootstrap.com/docs/4.5/layout/overview/), 


## :smile: Happy conding!

Ceejayski commented:
Please for the sticky navbar on the bigger screens, require the use of JavaScript which isn't needed in our project, we can make it sticky on mobile view but not on bigger screen as they are different navbars

sneeu-leeu committed:
Add style reset

sneeu-leeu committed:
Fix Newsletter section

sneeu-leeu commented:
Thank you Julio for your input and clear instructions as to inspect page size and breakpoints. 

This has been considered and used in the creation of this project.

#### To the next review:
### Please note that the project was built using DEFAULT bootstrap breakpoints and we attempted to learn using bootstrap without changing the values of the default bootstrap settings. Thus using media queries to select breakpoints for our page would be redundant for this project. Please review on bootstrap pixel widths or explain in more detail if we have misunderstood.  The sticky navbar in a large view is only achieved with technologies not used in this project, so we only fixed the small view.

Thanks!

##### After the previous review we have :
* Removed unnecessary files
* Fixed position issue with subscribe section
* separated reset and main CSS

umairarshadbutt commented:
##
## Pull Request under review by Umair Arshad (TSE)

umairarshadbutt commented on css/reset.css:
```diff
@@ -0,0 +1,141 @@
+}
+
+a {
+  color: inherit !important;
```
remove `!important`

umairarshadbutt commented on css/style.css:
```diff
@@ -0,0 +1,227 @@
+
+@media (min-width: 1200px) and (max-width: 1339px) {
+  .container-xl {
+    max-width: 97% !important;
```
remove `!important`

umairarshadbutt commented on css/style.css:
```diff
@@ -0,0 +1,227 @@
+
+@media (min-width: 1340px) {
+  .container-xl {
+    max-width: 1310px !important;
```
remove `!important`

umairarshadbutt reviewed (changes requested):
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @Ceejayski AND @Stephan 
Good job with the changes! :tada: We are almost there.
We, however, do have some few changes to carry out before we proceed (see review comment).

##

 For this project, we have to make sure nothing is messed up. Kindly compare the screenshots below 👇 and work on the suggested changes.
- [x] Remove the `!impoetant` from css file. ([Article for support](https://speckyboy.com/good-bad-css-practices/)) ( [screenshot](https://user-images.githubusercontent.com/29689273/93404399-8997b500-f8a3-11ea-8f88-fb12b8f6ca80.png) from Pathwright)

- [ ] Navbar is not sticky on some breakpoints
- [x] The debate section is not responsive according to the original page. for better understanding, I add the screenshots 👇

|[Source Page](https://www.newsweek.com/)|[Clone Page](https://ecstatic-snyder-879536.netlify.app/)|
|:---:|:---:|
|![n-o-debate](https://user-images.githubusercontent.com/29689273/93450401-5e3ab780-f8ef-11ea-8ded-5e930be05f0b.png)|![n-c-debate](https://user-images.githubusercontent.com/29689273/93450435-685cb600-f8ef-11ea-93a4-4f2cb1bbccd8.png)|

##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you stick at any point or have any questions, let me know on slack at any time. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

sneeu-leeu committed:
removed important commands where possible

sneeu-leeu commented:
## PLEASE READ BEFORE REVIEW:
#### The sticky navbar in a large view is only achieved with technologies not used in this project, so we only fixed the small view navbar to the top. the rest is redundant for this project.

#### We have:
* removed !important tags from style.css but not from the reset.css as it is overwritten by bootstrap commands.
* Ensured all position displays are equal on all sizes (as generalized by BOOTSTRAP breakpoints)

umairarshadbutt reviewed (approved):
# **Status: PROJECT APPROVED** :champagne:
Excellent job on this project guys. :fireworks: Keep it up.
##
### The [clone](https://ecstatic-snyder-879536.netlify.app/) shows a good understanding and usage of Bootstrap. :100:
##
- The cloned page is rendering as it should and meets all the project specifications.
- The project has a descriptive README.md and is professionally done. Good job with that.
- Pull Request has a meaningful title and a short description.
- CSS is clean. No linter errors at this point and no bad practices used as well.
- The markup as well has no errors nor warnings to show. Good job with that.
- The repo has followed all the guidelines there are to follow (GitHub flow etc.)
**All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete.**
- Congratulations on the splendid job @Stephan and @sneeu-leeu ! :+1: 
- You can now merge to master and submit the project completion form :rocket: . I wish you all the very best in your next tasks :1st_place_medal:
##
- **You used `!important` is recommended its usage, Which is the plus point :champagne:**
##
**If you have any questions, let me know on slack at any time. I will be happy to help.**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)
